contracts/base/OnlySelfCallable.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity ^0.8.18;
+
+abstract contract OnlySelfCallable {
+    error InvalidSender(address sender);
+    error InvalidCalldataLength();
+
+    modifier onlySelf() {
+        checkOnlySelf();
+        _;
+    }
+
+    function checkOnlySelf() private view {
+        if (msg.data.length < 20) {
+            revert InvalidCalldataLength();
+        }
+        address sender;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            sender := shr(96, calldataload(sub(calldatasize(), 20)))
+        }
+        if (sender != msg.sender) {
```

Comment:
too strict check here?

Reply:
This is will block relayed transactions

Reply:
Hmm why would it block this? The relaying would happen on Safe level not Manager level, right?

Reply:
Yes, if transaction is relayed to Safe and then Manager is called then no issues.